cpp/include/cudf/strings/find.hpp

```diff
@@ -0,0 +1,134 @@
+ * end of each string becomes the final position to include in the search.
+ *
+ * Any null string entries return corresponding null output column entries.
+ *
+ * @throw cudf::logic_error if start position is greater than stop position.
+ *
+ * @param strings Strings instance for this operation.
+ * @param target UTF-8 encoded string to search for in each string.
+ * @param start First character position to include in the search.
+ * @param stop Last position (exclusive) to include in the search.
+ *             Default of -1 will search to the end of the string.
+ * @param mr Resource for allocating device memory.
+ * @return New integer column with character position values.
+ */
+std::unique_ptr<column> find( strings_column_view const& strings,
+                              string_scalar const& target,
+                              size_type start=0, size_type stop=-1,
+                              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**
+ * @brief Returns a column of character position values where the target
+ * string is first found searching from the end of each string.
+ *
+ * If the string is not found, -1 is returned for that entry.
+ *
+ * The target string is searched within each string in the character
+ * position range [start,stop). If the stop parameter is -1, then the
+ * end of each string becomes the final position to include in the search.
+ *
+ * Any null string entries return corresponding null output column entries.
+ *
+ * @throw cudf::logic_error if start position is greater than stop position.
+ *
+ * @param strings Strings instance for this operation.
+ * @param target UTF-8 encoded string to search for in each string.
+ * @param start First position to include in the search.
+ * @param stop Last position (exclusive) to include in the search.
+ *             Default of -1 will search starting at the end of the string.
+ * @param mr Resource for allocating device memory.
+ * @return New integer column with character position values.
+ */
+std::unique_ptr<column> rfind( strings_column_view const& strings,
+                               string_scalar const& target,
+                               size_type start=0, size_type stop=-1,
+                               rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+/**
+ * @brief Returns a column of boolean values for each string where true indicates
+ * the target string was found within that string in the provided column.
+ *
+ * If a target string is not found, false is returned for that entry for that column.
+ *
+ * Any null string entries return corresponding null entries in the output columns.
+ *
+ * @param strings Strings instance for this operation.
+ * @param target UTF-8 encoded string to search for in each string.
+ * @param mr Resource for allocating device memory.
+ * @return New BOOL8 column.
+ */
+std::unique_ptr<column> contains( strings_column_view const& strings,
```

Comment:
There is a request for multi search: https://github.com/rapidsai/cudf/issues/3077  In other words a column of targets, and the output for a row i is true if `strings[i]` matches any row in `targets`. This will need to work for strings.

Reply:
`contains_any`, perhaps?